### DISCUSSION TITLE
Check if the other resources referenced by an object exist within a webhook

### DISCUSSION THREAD

**Discussion by zhixiongdu027** [kubernetes-sigs/kubebuilder#4199] [Support]
I want to check if the other resources referenced by an object exist within a webhook. Could someone please help me complete the logic in the following code example? 

Thank you very much.

```go
// SetupWebhookWithManager will setup the manager to manage the webhooks
func (r *Pool) SetupWebhookWithManager(mgr ctrl.Manager) error {
	return ctrl.NewWebhookManagedBy(mgr).
		For(r).
		Complete()
}

var _ webhook.Validator = &Pool{}

func (r *Pool) Validate() error {
	var allErrs field.ErrorList

	model := r.Spec.Model

	/*
		check model is exist here;
	*/

	if len(allErrs) == 0 {
		return nil
	}

	return apierrors.NewInvalid(schema.GroupKind{Group: GroupVersion.Group, Kind: "Pool"}, r.Name, allErrs)
}

// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
func (r *Pool) ValidateCreate() (admission.Warnings, error) {
	poollog.Info("validate create", "name", r.Name)

	// TODO(user): fill in your validation logic upon object creation.
	return nil, r.Validate()
}
```

**Comment by camilamacedo86** [comment]
Sorry, but could you please provide more info.

> I want to check if the other resources referenced by an object exist within a webhook

Referenced how? if OwnerRef, via labels could you please clarify?

> Could someone please help me complete the logic in the following code example?

what do you exactly want to do in your example?
What logic do you need?
What are you trying to achieve?

**Comment by camilamacedo86** [comment]
Closing as no further engagement from the author’s side. However, please feel free to re-open if you still need assistance.
